--- a/libc/sysdeps/linux/powerpc/bits/syscalls.h
+++ b/libc/sysdeps/linux/powerpc/bits/syscalls.h
@@ -128,38 +128,38 @@
 # define LOAD_ARGS_1(name, __arg1) \
 	long int arg1 = (long int) (__arg1); \
 	LOAD_ARGS_0(name, 0); \
-	if (__builtin_classify_type (__arg1) != 5 && sizeof (__arg1) > 4) \
+	if (__builtin_classify_type (__arg1) != 5 && sizeof (__arg1) > sizeof(long)) \
 		__illegally_sized_syscall_arg1 (); \
 	r3 = arg1
 # define LOAD_ARGS_2(name, __arg1, __arg2) \
 	long int arg2 = (long int) (__arg2); \
 	LOAD_ARGS_1(name, __arg1); \
-	if (__builtin_classify_type (__arg2) != 5 && sizeof (__arg2) > 4) \
+	if (__builtin_classify_type (__arg2) != 5 && sizeof (__arg2) > sizeof(long)) \
 		__illegally_sized_syscall_arg2 (); \
 	r4 = arg2
 # define LOAD_ARGS_3(name, __arg1, __arg2, __arg3) \
 	long int arg3 = (long int) (__arg3); \
 	LOAD_ARGS_2(name, __arg1, __arg2); \
-	if (__builtin_classify_type (__arg3) != 5 && sizeof (__arg3) > 4) \
+	if (__builtin_classify_type (__arg3) != 5 && sizeof (__arg3) > sizeof(long)) \
 		__illegally_sized_syscall_arg3 (); \
 	r5 = arg3
 # define LOAD_ARGS_4(name, __arg1, __arg2, __arg3, __arg4) \
 	long int arg4 = (long int) (__arg4); \
 	LOAD_ARGS_3(name, __arg1, __arg2, __arg3); \
-	if (__builtin_classify_type (__arg4) != 5 && sizeof (__arg4) > 4) \
+	if (__builtin_classify_type (__arg4) != 5 && sizeof (__arg4) > sizeof(long)) \
 		__illegally_sized_syscall_arg4 (); \
 	r6 = arg4
 # define LOAD_ARGS_5(name, __arg1, __arg2, __arg3, __arg4, __arg5) \
 	long int arg5 = (long int) (__arg5); \
 	LOAD_ARGS_4(name, __arg1, __arg2, __arg3, __arg4); \
-	if (__builtin_classify_type (__arg5) != 5 && sizeof (__arg5) > 4) \
+	if (__builtin_classify_type (__arg5) != 5 && sizeof (__arg5) > sizeof(long)) \
 		__illegally_sized_syscall_arg5 (); \
 	r7 = arg5
 # define LOAD_ARGS_6(name, __arg1, __arg2, __arg3, __arg4, __arg5, __arg6) \
 	long int arg6 = (long int) (__arg6); \
 	LOAD_ARGS_5(name, __arg1, __arg2, __arg3, __arg4, __arg5); \
-	if (__builtin_classify_type (__arg6) != 5 && sizeof (__arg6) > 4) \
+	if (__builtin_classify_type (__arg6) != 5 && sizeof (__arg6) > sizeof(long)) \
 		__illegally_sized_syscall_arg6 (); \
 	r8 = arg6
