--- a/scripts/build/arch/powerpc.sh	2026-01-31 09:08:40.674803776 -0500
+++ b/scripts/build/arch/powerpc.sh	2026-01-31 09:08:55.078947536 -0500
@@ -19,6 +19,11 @@
             ;;
     esac
 
+
+    if [ "${CT_ARCH_FLOAT}" = "soft" ]; then
+        CT_TARGET_SYS="${CT_TARGET_SYS}sf"
+    fi
+
     # Add extra flags for SPE if needed. SPE is obsolete in GCC8.
     if [ "${CT_ARCH_powerpc_ABI_SPE}" = "y" ]; then
         CT_ARCH_TARGET_CFLAGS="-mabi=spe -mspe"
